gradle-baseline-java/src/test/groovy/com/palantir/baseline/BaselineExactDependenciesTest.groovy

```diff
@@ -109,6 +109,32 @@ class BaselineExactDependenciesTest extends AbstractPluginTest {
         BuildResult result = with('checkUnusedDependencies', '--stacktrace').build()
         result.task(':classes').getOutcome() == TaskOutcome.SUCCESS
         result.task(':checkUnusedDependencies').getOutcome() == TaskOutcome.SUCCESS
+        result.task(':checkUnusedDependenciesMain').getOutcome() == TaskOutcome.SUCCESS
+    }
+
+    def 'checkUnusedDependenciesTest passes if dependency from main source set is not referenced in test'() {
+        when:
+        buildFile << standardBuildFile
+        buildFile << """
+        repositories {
+            mavenCentral()
+        }
+        dependencies {
+            compile 'com.google.guava:guava:28.0-jre'
+        }
+        """
+        file('src/main/java/pkg/Foo.java') << '''
+        package pkg;
+        public class Foo {
+            void foo() {
+                com.google.common.collect.ImmutableList.of();
+            }
+        }
+        '''.stripIndent()
+
+        then:
+        def result = with('checkUnusedDependencies', '--stacktrace').build()
```

Comment:
Without my latest changes (with the previous implementation at https://github.com/palantir/gradle-baseline/pull/1262/files/bc8e9e2cca856833a02d3d8775c9641378a8953f) this test failed with:

```
* What went wrong:
Execution failed for task ':checkUnusedDependenciesTest'.
> Found 1 dependencies unused during compilation, please delete them from 'build.gradle' or choose one of the suggested fixes:
  	com.google.guava:guava
```